internal/command/machine/clone.go

```diff
@@ -159,15 +160,34 @@ func runMachineClone(ctx context.Context) (err error) {
 		fmt.Fprintf(io.Out, "Auto destroy enabled and will destroy machine on exit. Use --clear-auto-destroy to remove this setting.\n")
 	}
 
-	if volID := flag.GetString(ctx, "attach-volume"); volID != "" {
-		if len(source.Config.Mounts) != 1 {
-			return fmt.Errorf("Can't attach the volume as the source machine doesn't have any volumes configured")
+	var volID string
+	if volumeInfo := flag.GetString(ctx, "attach-volume"); volumeInfo != "" {
+		splitVolumeInfo := strings.Split(volumeInfo, ":")
+
+		if len(source.Config.Mounts) > 1 {
+			return fmt.Errorf("Can't use --attach-volume for machines with more than 1 volume.")
+		} else if len(source.Config.Mounts) == 1 && len(splitVolumeInfo) > 1 {
+			return fmt.Errorf("Can't set a mount path on a machine with a volume, please use only the volume id on '%s'", volumeInfo)
+		} else if len(source.Config.Mounts) == 0 && len(splitVolumeInfo) != 2 {
+			return fmt.Errorf("Couldn't find a mount path on '%s'", volumeInfo)
+		}
+
+		// patch the source config so the loop below attaches the volume on the passed mount path
+		if len(source.Config.Mounts) == 0 && len(splitVolumeInfo) == 2 {
+			volID = splitVolumeInfo[0]
```

Comment:
`volID = splitVolumeInfo[0]` can be factored out of the `if` block and then you can factor even more when `len(source.Config.Mounts) == 0`

Reply:
good catch